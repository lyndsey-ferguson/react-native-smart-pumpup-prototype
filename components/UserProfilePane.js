'use strict'

import React, {
  Component
} from 'react';
import {
  Dimensions,
  FlatList,
  Image,
  StyleSheet,
  Text,
  View
} from 'react-native';

const { width: windowWidth } = Dimensions.get('window');
const imagePreviewPaneWidth = windowWidth - 40;

export default class UserProfilePane extends React.Component {
  renderImagePreview({item}) {
    return (
      <Image style={{width: imagePreviewPaneWidth, height: imagePreviewPaneWidth}} source={item.source} key={item.key}/>
    )
  }
  render() {
    const { isLoading, photo, name, bio } = this.props;
    const imageSource = isLoading ? require('../res/no_avatar.png') : photo;
    const images = [
      {
        key: 'me 1',
        source: require('../res/me.png')
      },
      {
        key: 'me 2',
        source: require('../res/me.png')
      }
    ];
    return (
      <View style={{flex: 1, flexDirection: 'column', backgroundColor: '#f00'}}>
        <View style={styles.container}>
          <Image style={styles.avatar}
                 resizeMode={'contain'}
                 source={imageSource} />
          <View style={styles.infoPane}>
            <Text style={styles.name}>{name}</Text>
            <Text numberOfLines={3}
                  ellipsizeMode={'tail'} >
              {bio}
            </Text>
          </View>
        </View>
        <FlatList data={images}
                  style={styles.imagePreviews}
                  horizontal={true}
                  decelerationRate={0}
                  snapToAlignment={'center'}
                  snapToInterval={imagePreviewPaneWidth}
                  renderItem={this.renderImagePreview.bind(this)} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: '#fff',
    borderBottomWidth: 1.0,
    alignItems: 'flex-end',
    borderColor: '#444',
    paddingBottom: 5,
    paddingTop: 5
  },
  avatar: {
    width: 80,
    height: 80
  },
  imagePreviews: {
    width: imagePreviewPaneWidth,
    height: imagePreviewPaneWidth,
    marginTop: 20,
    marginLeft: 20
  },
  infoPane: {
    flex: 1,
    marginLeft: 5
  },
  name: {
    fontWeight: 'bold',
    fontSize: 20
  }
});
