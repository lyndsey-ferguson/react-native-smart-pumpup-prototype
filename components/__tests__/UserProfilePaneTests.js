import React from 'react';
import UserProfilePane from '../UserProfilePane';
import renderer from 'react-test-renderer';

it('renders correctly without an avatar', () => {
  const tree = renderer.create(
    <UserProfilePane name={'Reg Park'} bio={'Likes to lift weights and do yoga #YogaBear'}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with an avatar', () => {
  const tree = renderer.create(
    <UserProfilePane avatarUri={'https://www.example.com/avatar.png'} name={'Reg Park'} bio={'Likes to lift weights and do yoga #YogaBear'}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
