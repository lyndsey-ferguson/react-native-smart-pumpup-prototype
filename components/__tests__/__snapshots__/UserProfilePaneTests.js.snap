// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with an avatar 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f00",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#444",
        "flexDirection": "row",
        "paddingBottom": 5,
        "paddingTop": 5,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "https://www.example.com/avatar.png",
        }
      }
      style={
        Object {
          "height": 80,
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Reg Park
      </Text>
      <View
        linkDefault={true}
        linkText={[Function]}
        onPress={[Function]}
        style={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={0}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            >
              Likes to lift weights and do yoga #YogaBear
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly without an avatar 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f00",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#444",
        "flexDirection": "row",
        "paddingBottom": 5,
        "paddingTop": 5,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 80,
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Reg Park
      </Text>
      <View
        linkDefault={true}
        linkText={[Function]}
        onPress={[Function]}
        style={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={0}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            >
              Likes to lift weights and do yoga #YogaBear
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
